agent/consul/config_replication.go

```diff
@@ -30,7 +31,13 @@ func diffConfigEntries(local []structs.ConfigEntry, remote []structs.ConfigEntry
 		if configentry.EqualID(local[localIdx], remote[remoteIdx]) {
 			// config is in both the local and remote state - need to check raft indices
 			if remote[remoteIdx].GetRaftIndex().ModifyIndex > lastRemoteIndex {
-				updates = append(updates, remote[remoteIdx])
+				// if ID is not available in either configs, replicate it
+				if remote[remoteIdx].ID() == "" || local[localIdx].ID() == "" {
+					updates = append(updates, remote[remoteIdx])
+					// if ID is different, replicate it
+				} else if remote[remoteIdx].ID() != local[localIdx].ID() {
+					updates = append(updates, remote[remoteIdx])
+				}
```

Comment:
This is the main change from replication perspective, we still use the index to track but now if we need to write a config entry we check that the version we have locally is not the same as the one we are writing (based on the ULID).